src/Handlers/Reply/ReplyHandler.ts

```diff
@@ -0,0 +1,71 @@
+import crypto from "crypto";
+import HttpMethods from "../../Constants/HttpMethods";
+import {ICredentials} from "../../Utils/Types";
+import {Hmac} from "../../Request/Hmac";
+import buckarooClient from "../../BuckarooClient";
+
+export class ReplyHandler {
+    private readonly data: object
+    private readonly uri?: string
+    private readonly auth_header?: string
+    private credentials: ICredentials;
+    private _isValid: boolean = false
+
+    constructor(credentials: ICredentials, data: string | object,auth_header?: string, uri?: string) {
+        if(typeof data === 'string'){
+            try {
+                this.data = JSON.parse(data)
+            } catch (e){
+                let objData = {}
+                new URLSearchParams(data).forEach((value, name)=>{
+                    objData[name] = value
+                })
+                this.data = objData
+            }
+        }else {
+            this.data = data
+        }
+        this.credentials = credentials
+        this.uri = uri
+        this.auth_header = auth_header
+    }
+    get isValid(){
+        return this._isValid
+    }
+    validate() {
+        if(this.data["Key"] && this.auth_header && this.uri){
+            this._isValid = this.validateJson(this.auth_header)
+        }else if (this.data["brq_signature"]){
+            this._isValid = this.validateHttp({...this.data})
+        }else {
+            throw new Error('Invalid reply data')
+        }
+        return this
+    }
```

Comment:
Please try to limit your if-else statement

```javascript
validate() {
        if(this.data["Key"] && this.auth_header && this.uri) {
            this._isValid = this.validateJson(this.auth_header)
return this
        }

if (this.data["brq_signature"]){
            this._isValid = this.validateHttp({...this.data})
return this
        }
       
 throw new Error('Invalid reply data')
    }
```

This way is much easier to read.